### DISCUSSION TITLE
Some problems with parallelization

### DISCUSSION THREAD

**Discussion by zhaowb12138** [SSCHAcode/python-sscha#101] [Q&A] [answered]
Hi developers,
I have some problems with parallelization. I want to use parallelization to calculate spectral function. 
It's my calculate scrip.
[plot_fre.txt](https://github.com/SSCHAcode/python-sscha/files/9173075/plot_fre.txt)

This is the script to submit the task to the cluster. 
[sub.txt](https://github.com/SSCHAcode/python-sscha/files/9173077/sub.txt)

Parallelization doesn't work. It will work 48 times. Is there something wrong? And what should I do ?

**Comment by zhaowb12138** [comment]
oh，I have another question can get_free_energy_hessian()parallelization？When I make include_v4 = True,it works really slow.
Looking forward to your reply,thank you.
Zhaowb

**Reply by mesonepigreco** [reply]
Hi Zhaowb, 
Are you using the most updated version of Cellconstructor from the github master repository?  I remember there was some issue with the parallelization, probably has been fixed now in one of the branches.
 @josudiego @DjordjeDangic @diegomartinez2   do you know how to use the parallelized version, can we merge this into the master?

Regarding the parallelization of the free energy hessian with the fourth order, it is a work in progress, as we are reworking the algorithm to reduce the scaling size and the memory requirements.

**Comment by diegomartinez2** [comment, marked as answer]
Hi, my implementation is not in main, but you can find it in [pull request](https://github.com/SSCHAcode/CellConstructor/pull/45). With that implementation you can use 'get_static_correction_along_path_multiprocessing' that uses the same input as 'get_static_correction_along_path' but with the extra input "processes = N" .

> processes     : integer
>                        Number of maximum processes (cpu) to be used during the calculation.
>                        If 'None' then the number returned by os.cpu_count() is used.
>                        (default: None)

This is probably not the best solution for you, but I hope this helps.
Note that the multiprocess implementation is still in works for the "get_diag_dynamic_correction_along_path_multiprocessing" function.
